### PR TITLE
pkg-config support

### PR DESCRIPTION
**Pull request by PetrovKP**
Add generator of pkg-configs for configurations for linux, windows, mac OS:
- dal-dynamic-sequential-host
- dal-dynamic-threading-host
- dal-static-sequential-host
- dal-static-threading-host

For dpc support will update in next PR.

Example generator for `dal-dynamic-threading-host`
```pc
#===============================================================================
# Copyright 2021 Intel Corporation.
#
# This software and the related documents are Intel copyrighted  materials,  and
# your use of  them is  governed by the  express license  under which  they were
# provided to you (License).  Unless the License provides otherwise, you may not
# use, modify, copy, publish, distribute,  disclose or transmit this software or
# the related documents without Intel's prior written permission.
#
# This software and the related documents  are provided as  is,  with no express
# or implied  warranties,  other  than those  that are  expressly stated  in the
# License.
#===============================================================================

prefix=${pcfiledir}/../../
exec_prefix=${prefix}
libdir=${exec_prefix}/lib/intel64
includedir=${prefix}/include

#info
Name: oneDAL
Description: Intel(R) oneAPI Data Analytics Library
Version: [dal.version]
URL: https://software.intel.com/en-us/oneapi/onedal
#Link line
Libs: ${libdir}/libonedal.so ${libdir}/libonedal_core.so ${libdir}/libonedal_thread.so -ltbb -ltbbmalloc -lpthread -ldl
#Compiler line
Cflags: -std=c++17 -I${includedir}
```

### CODE REVIEW

**Comment by PetrovKP** [comment]
http://intel-ci.intel.com/ebd938a3-dd7e-f1a9-a7c8-8cdcd4b723a1

File: deploy/local/vars_lnx.sh [hunk 1]
```diff
@@ -230,3 +230,4 @@ export LIBRARY_PATH=$__daal_tmp_dir/lib/intel64${LIBRARY_PATH+:${LIBRARY_PATH}}
 export LD_LIBRARY_PATH=$__daal_tmp_dir/lib/intel64${LD_LIBRARY_PATH+:${LD_LIBRARY_PATH}}
 export CLASSPATH=$__daal_tmp_dir/lib/onedal.jar${CLASSPATH+:${CLASSPATH}}
 export CMAKE_PREFIX_PATH=$__daal_tmp_dir${CMAKE_PREFIX_PATH+:${CMAKE_PREFIX_PATH}}
+export PKG_CONFIG_PATH=$__daal_tmp_dir/lib/pkgconfig${PKG_CONFIG_PATH+:${PKG_CONFIG_PATH}}
```
Comment (napetrov) [inline deploy/local/vars_lnx.sh:233]: Think this also should be exposed via modulefiles

Comment (PetrovKP) [reply deploy/local/vars_lnx.sh:233]: thanks, added

**Comment by PetrovKP** [comment]
http://intel-ci.intel.com/ebd99b68-c485-f179-bc1d-8cdcd4b723a1

**Comment by napetrov** [comment]
Looks good overall, but have some problems in internal CI

File: deploy/local/vars_lnx.sh (hunk 1 above)
Comment (PetrovKP) [reply deploy/local/vars_lnx.sh:233]: I clarified, now this is not required in the framework oneAPI libraries

**Comment by PetrovKP** [comment]
http://intel-ci.intel.com/ebda7ab3-9f9f-f1c7-88bd-8cdcd4b723a1

**Review by rlnx — APPROVED** [review]
I approve assuming testing in CI will be added in few weeks

**Pull request merged**